src/UkModulusCheck/Implementation.fs

```diff
@@ -0,0 +1,232 @@
+                        // Perform the second double alternate check, and for the double alternate check with exception 5 the checkdigit is h from the original account number, except:
+                        // After dividing the result by 10:
+                        // - if the remainder = 0 and h = 0 the account number is valid
+                        // - for all other remainders, take the remainder away from 10. If the number you get is the same as h then the account number is valid.
+                        match doubleAlternateModulus rule2.Weightings number' % 10 with
+                        | 0 when number'.[Position.H] = '0' -> true
+                        | reminder -> (10 - reminder) = char2int number'.[Position.H]
+
+                    firstCheck() && secondCheck()
+
+                | Exception 6, Exception 6 ->
+                    // Indicates that these sorting codes may contain foreign currency accounts which cannot be checked. Perform the first and second checks, except:
+                    // If a = 4, 5, 6, 7 or 8, and g and h are the same, the accounts are for a foreign currency and the checks cannot be used.
+                    if ("45678" |> Seq.contains number.[Position.A]) && number.[Position.G] = number.[Position.H]
+                    then true
+                    else validateRule rule1 number && validateRule rule2 number
+
+                | Exception 10, Exception 11 ->
+                    // These exceptions are for some Lloyds accounts and some TSB accounts. If there is a 10 in the exception column for the first check for a sorting code
+                    // and an 11 in the exception column for the second check for the same sorting code, if either check is successful the account number is deemed valid.
+
+                    let firstCheck() =
+                        // For the exception 10 check, if ab = 09 or ab = 99 and g = 9, zeroise weighting positions u-b.
+                        let rule1' =
+                            if ["09"; "99"] |> List.contains number.[Position.A..Position.B] && number.[Position.G] = '9'
+                            then { rule1 with Weightings = [for i, w in rule1.Weightings |> List.indexed -> if i >= Position.U && i <= Position.B then 0 else w] }
+                            else rule1
+                        validateRule rule1' number
+
+                    let secondCheck() =
+                        validateRule rule2 number
+
+                    firstCheck() || secondCheck()
+
+                | Exception 12, Exception 13 ->
+                    // Where there is a 12 in the exception column for the first check for a sorting code and a 13 in the exception column for the second check for the same sorting code,
+                    // if either check is successful the account number is deemed valid.
+                    validateRule rule1 number || validateRule rule2 number
+
+                | _ -> false // TODO: Should we fail for unrecognized exception?
```

Comment:
I think we may want to return at least some information in `Invalid` case not just `Valid/Invalid`

Reply:
Something like `Valid | Invalid of Reason and Reason = ValidationFailed | UnrecognizedException | ...`?